src/api/libcellml/component.h

```diff
@@ -210,6 +213,89 @@ class LIBCELLML_EXPORT Component: public ComponentEntity
      */
     bool hasVariable(const std::string &name) const;
 
+    /**
+     * @brief Add a reset by reference as part of this component.
+     *
+     * Add a reset by reference as part of the given component.
+     *
+     * @sa removeReset
+     *
+     * @param r The reset to add.
+     */
+    void addReset(const ResetPtr &r);
```

Comment:
For consistency, we might want to replace `r` with `reset` (as is the case with `removeReset()` below).